Update e2e testing documentation

worldomonation opened the pull request:
#### Changes proposed in this Pull Request

This PR addresses the testing documentation update task, following the outline posted in pciE2j-aW-p2.

`README.md`
* add new entries in `README.md` table of contents that link to the new documentation pages.
* remove existing content from page as they are addressed by the new documents.

`setup.md`
* migrate environment setup process to this document and expand on it.
* migrate details on configuration files, and point to internal FG page for the process.
* migrate details about environment variables and branch naming strategy to this document.

`environment_variables.md`
* new document split from the old `config.md`.

`config_values.md`
* new document split from the old `config.md`.

`overview.md`
* a high-level overview that outlines the goals of quality squad and e2e testing.

`running-tests.md`
* refactor existing document.
* covers multiple scenarios of developer running tests.

`style-guide.md`
* refactor of existing document.
* add discussion on tags (parallel, jetpack, etc)

`troubleshooting_debugging.md`
* discussion of possible issue faced by those that pulled the repo fresh.

`writing_tests.md`
* new document outlining some major 'components' to consider when writing tests.
* migrated content from old `gutenberg.md` to become a section on this document.

#### Testing instructions

I have submitted my work in my forked repo of `wp-calypso`.

Please go to https://github.com/worldomonation/wp-calypso/tree/update/e2e-documentation/test/e2e to try it out.

Fixes #

worldomonation committed:
Changes:
Begin initial update of the e2e test documentation.

README.md

* new table of contents with new structure to replace existing one.

config_values.md

* moved configuration values rundown to a separate page dedicated to it.
* sorted the attributes by alphabetical order.

setup.md

* new page detailing anything to do with the environment setup process for e2e tests.
* include content from current `config.md`, main README.md.
* addition of new content in `config_values.md`.

worldomonation committed:
Changes:
Include links in the top `README.md` table of contents that links to the various sections of newly added `setup.md` file.

cathymcbride commented on test/e2e/README.md:
```diff
@@ -4,14 +4,12 @@ Automated end-to-end acceptance tests for the [wp-calypso](https://github.com/Au
+  - [configuration file](https://github.com/worldomonation/wp-calypso/blob/update/e2e-documentation/test/e2e/docs/setup.md#configuration)
+  - [environment variables](https://github.com/worldomonation/wp-calypso/blob/update/e2e-documentation/test/e2e/docs/setup.md#environment-variables)
+  - [branch naming](https://github.com/worldomonation/wp-calypso/blob/update/e2e-documentation/test/e2e/docs/setup.md#naming-branches)
+  - [set up user (optional)]()
```
Should this have a link top something specific?

worldomonation replied:
Yes - I had not filled out the content of `set up user` yet and so I had left it blank. 

It is not something I've done yet, so my plan is to transplant the appropriate content of the FG page (PCYsg-jvX-p2) here in a subsequent push.

worldomonation replied:
I've decided to remove the subsection on `set up user` because it references some internal links, so would be much better suited to a consolidated page on FG.

cathymcbride reviewed (dismissed):
One tiny thing to look at, but overall I think it look great!

worldomonation commented:
> One tiny thing to look at, but overall I think it look great!

Excellent. I am in the process of updating other sections as I write this, but the content scope is larger than I expected it to be. I will try to reuse as much of the existing documentation where applicable to timebox this work.

worldomonation committed:
Changes:
Rewrite article on running tests on various environments.

`runnig-tests.md`
* update sections and table of contents.
* refactor existing content to remove instructions on environment setup as they are referenced in the `setup.md` file.
* reorganize content to three environments: local, CircleCI, Sauce Labs.
* expand on each section and populate with real-world examples.

WIP
* CircleCI docker image is a work in progress.

worldomonation committed:
Changes:
Update TOC listings for top-level `README.md` fie.

`README.md`
* update the links under `Setup` as it was referring to a URL of the forked repo.
* insert updated TOC for `Running Tests` and associated links.

worldomonation committed:
Changes:
Include more details on decrypting secrets.

`setup.md`
* transplant steps from internal documentation about how to decrypt secrets.
* rephrase some statements under Custom Configurations

worldomonation committed:
Changes:
Use separate file for tracking environment variables and include a new Overview section.

`environment_variables.md`
* move the contents of `test/e2e/docs/config.md#standalone-environment-variables` to a new standalone file.

`overview.md`
* add new overview section to discuss at a high level the objectives of e2e testing, technology used and the main targets for e2e tests.

`running-tests.md`
* include schedule information (incomplete).

worldomonation committed:
Changes:
Fix the ToC for `setup.md` and `running-tests.md`.

bsessions85 commented on test/e2e/docs/setup.md:
```diff
@@ -0,0 +1,119 @@
+
+Follow these steps to decrypt the configuration file:
+
+1. obtain the secret key from A8C Secret Store.
```
I would go into less detail here. maybe just refer internal users to the field guide

worldomonation replied:
Got it - I will move this entire section to the FG and on this page, simply make a note about referring to that page.

worldomonation replied:
Removed with subsequent pushes.

bsessions85 commented on test/e2e/docs/running-tests.md:
```diff
@@ -2,112 +2,168 @@
+
+### Full suite
+
+The following labels provide ability to trigger e2e tests for specific feature(s):
```
I am not sure if we should put this here or not. I guess it doesn't matter, but these won't work for external users

worldomonation replied:
If it does not have to strictly remain internal, I would prefer to leave here - would prefer to see GitHub documentation be the ongoing, living documentation of e2e tests and this would align with the goal.

With that said, I am open to moving this to a consolidated FG page as well.

bsessions85 replied:
I'm fine with leaving it.

worldomonation committed:
Changes:
Add content on writing tests and style guides.

*setup.md*
* remove content relating to decrypting the configuration file.

*style-guide.md*
* insert table of content to the top of page.
* updated the example on constructing page objects.

`writing_tests.md`
* add new page with details on page objects, flows, components and selectors.

worldomonation committed:
Changes:
Update the main page table of contents, write troubleshooting/debuggin page.

`README.md`
* update the table of contents.
* remove remaining content of the page.

`trobuleshooting_debugging.md`
* add page and content about issues I have faced (eg. git pre-commit hooks)

`setup.md`
* refer to the troubleshooting guide.

worldomonation committed:
Changes:
Minor formatting update and removing `set up user` section.

worldomonation commented:
I've completed updating all sections on GitHub.

Some items are not as fleshed out as I would like 
eg. information on canary, writing e2e tests themselves.

But I think this structure will form a solid foundation onto which we can continue expanding on.

worldomonation committed:
Changes:
Correct dead links and add section on tags.

worldomonation committed:
Changes:
Added section on Apple Silicon and fixed dead links.

`README.md`
* fixed dead links.
* standardized on capitalization.

`troubleshooting.md`
* renamed file from `troubleshooting_debugging.md` as it was unnecessarily long.

`setup.md`
* added section on setting up Apple Silicon environment.

worldomonation commented:
I am ready to merge this in once approved! I feel like I've covered most angles, and if anything else comes up short it can be added later.

bsessions85 reviewed (approved):
This looks really good! We can always adjust where needed later.

worldomonation commented:
Closing this as I've pushed the branch to this repository itself (instead of a fork). Corresponding PR is #49496.
.
